docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,125 @@
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+
+  * the device thin-edge is running on is referred as the **main-device**
+    * thin-edge on the **main-device** establishes and manages all communication to the cloud
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device is represented in the cloud with an individual **device twin**
+  * a unique **child-id** makes the association between each **external child-device** and it's **device twin**
+  * all telemetry data (i.E. Measurements, Variables, Events, Alarms) of a device (i.E. **main-device** or **external child-devices**) appear
+    in the cloud in context of its corresponding **device twin**
+
+The figure below illustrates the device concept.
+
+![Device Concept](images/device-concept.svg)
+
+## thin-edge plugin concept
+
+**thin-edge** realizes full-fledged cloud functionality (e.g. _configuration management_ or _log management_) based on **plugins**.
+  * a **plugin** encapsulates and manages access to _ressources_ and _services_ of the device, as e.g.
+      * _software management_ accesses the device's _package manager_
+      * _configuration management_ accesses device's _configuration files_
+  * a **plugin** can be
+      * an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+      * or a thin-edge built-in software component (e.g. as for _software management_)
+  * usually a **plugin** runs on the **main-device**; thus it can access the _resources_ of the **main-device** directly
+  * to access _ressources_ of an **external child-device** a **plugin** needs another component, referred as **child-device agent**
+
+The figure below illustrates the concept of **plugins** and **child-devices agents**.
+
+![Plugin Concept](images/plugin-concept.svg)
+
+### Child-Device Agent
+  * a **child-device agent** is the counterpart of a **plugin**, that takes the responsibility to access to the **external child-device's** _resources_
+  * a **child-device agent** can serve one or more **plugins**
+  * a **child-device agent** can be installed and executed on the **external child-device**, or on the **main-device**
+    * if it runs in the **external child-device** it can access the _resources_ directly
+    * if it runs on the **main-device** it can use any (low-level) interfaces the **external child-device** provides to access those resources
+      * One main reason to install the **child-device agent** on the **main-device** is, when the **external child-device** cannot or shall not be altered.
+
+### Plugin-Contract
+
+A **plugin** defines and implements a specific **contract** for all interactions with a **child-device agent**
+  * part of the **contract** could be e.g.:
```

Comment:
Are we expecting it to be valid contract for for main device too?( forget about child devices)...so anyone can add a diferent implementation for the main device e.g. one can implement a different config management locally but has to adhere to the contract.

Reply:
Of course, we should consider that. Especially in a containerized environment it allows a plugin to access resources in another container via that contract.

Unfortunately that's not clear from the current document. I will add that.

Reply:
I added that aspect, see https://github.com/cstoidner/thin-edge.io/commit/0002a92a91a0f0bc7420a8fcdecd670a9ce6e13c